<?xml version="1.0"?>
<launch>
	<!-- these are the arguments you can pass this launch file, for example paused:=true -->
	  <arg name="model" default="$(find assignment2_exprob)/urdf/robot4.xacro" />
	<!-- Load the URDF into the ROS Parameter Server -->
	  <param name="robot_description" command="$(find xacro)/xacro --inorder $(arg model)" />
	  <param name="use_gui" value="True"/>
	<!-- Start Rviz -->  
	  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" /> 
	  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find assignment2_exprob)/config/rviz_assignment2.rviz" />
	<!-- Run a python script to the send a service call to gazebo_ros to spawn a URDF robot -->
	  <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen"
		args="-urdf -model robot_camera_ros1 -param robot_description -x 2.0 -y 1.0 -z 0.05"/> 
		
	  <rosparam file="$(find assignment2_exprob)/config/camera_config.yaml" command="load" ns="/robot4_xacro" />
	  
	  <node name="controller_spawner" pkg="controller_manager" type="spawner" output="screen" args="--namespace=/robot4_xacro
		    joint_state_controller
		    camera_velocity_controller
		    --shutdown-timeout 3">
	  </node>  

</launch>
